modules/crud/schema.go

```diff
@@ -132,7 +141,133 @@ func getFieldType(fieldType ast.Type, fieldTypeStuct *SchemaFieldType, doc *ast.
 	return nil
 }
 
+func (m *Module) schemaValidator(collectionFields SchemaField, doc map[string]interface{}) (map[string]interface{}, error) {
+	mutatedDoc := map[string]interface{}{}
+
+	for fieldKey, fieldValue := range collectionFields {
+		// check if key is required
+		value, ok := doc[fieldKey]
+		if fieldValue.IsFieldTypeRequired {
+			if !ok {
+				return nil, errors.New("Field " + fieldKey + " Not Present")
+			}
+		}
+
+		// check type
+		val, err := m.checkType(value, fieldValue)
+		if err != nil {
+			return nil, err
+		}
+		mutatedDoc[fieldKey] = val
+	}
+
+	return mutatedDoc, nil
+}
+
 // ValidateSchema checks data type
-func (m *Module) ValidateSchema() {
+func (m *Module) ValidateSchema(dbType, col string, req *model.CreateRequest) error {
+
+	if m.schema == nil {
+		return errors.New("Schema not initialized")
+	}
+
+	v := make([]map[string]interface{}, 0)
+
+	switch t := req.Document.(type) {
+	case []map[string]interface{}:
+		v = t
+	case map[string]interface{}:
+		v = append(v, t)
+	}
+
+	collection, ok := m.schema[dbType]
+	if !ok {
+		return errors.New("No db was found named " + dbType)
+	}
+	collectionFields, ok := collection[col]
+	if !ok {
+		return errors.New("No collection or table was found named " + col)
+	}
+
+	for index, fields := range v {
```

Comment:
Rename `fields` to `doc`

Reply:
You'll need to type assert `doc` to `map[string]interface{}`

Reply:
> 
> 
> You'll need to type assert `doc` to `map[string]interface{}`

why

Reply:
Because `v` is `[]interface{}` nor `[]map[string]interface{}`

Reply:
> 
> 
> You'll need to type assert `doc` to `map[string]interface{}`

should i handle panic error if assertion is not possible

Reply:
nope